server/engine/src/main/java/org/eclipse/lsp/cobol/service/copybooks/CopybookServiceImpl.java

```diff
@@ -286,12 +286,15 @@ public void sendCopybookDownloadRequest(String documentUri, Collection<String> c
               .flatMap(Set::stream)
               .collect(toList());
       LOG.debug("Copybooks to download: {}", copybooksToDownload);
+      List<CopyBookDTO> copybooks = new ArrayList<>();
+      copybooksToDownload.forEach(copybookName -> copybooks.add(new CopyBookDTO(copybookName.getDisplayName(), copybookName.getDialectType())));
+
       if (!copybooksToDownload.isEmpty()) {
         clientProvider.get().downloadCopybooks(documentUri,
-            copybooksToDownload.stream().map(CopybookName::getQualifiedName).collect(toList()),
-            Optional.ofNullable(copybooksToDownload.stream().findFirst().get().getDialectType()).orElse(COBOL), //NOSONAR
+                copybooks,
```

Comment:
Would something like this work? assuming a new constructor (or "from" function) would be created, that could also handle the possibility of dialect being `null` and replaced with the "COBOL".
```suggestion
	            copybooksToDownload.stream().map(CopyBookDTO::new).collect(toList()),
```

Reply:
If a dialect being null then "COBOL" should be the default value. Why do we wanna replace something is working well and with something same ?

Reply:
That's what should happen, but your code just passes in null.

Reply:
It was setting as default elsewhere but this method seems better wrt tests were failing 👍 . should resolve : b1f53f18620de7021eb4fa59f051336afa4b7368